src/main.ts

```diff
@@ -0,0 +1,128 @@
+const core = require ('@actions/core');
+const path = require('path');
+const https = require('https');
+
+const jsonPath = path.join(__dirname, '..', 'resources', 'notification.json');
+const workflow = process.env.GITHUB_WORKFLOW;
+const repository = process.env.GITHUB_REPOSITORY;
+const branch = process.env.GITHUB_REF;
+const event_payload =  process.env.GITHUB_EVENT_PATH || jsonPath;
+let event_name = process.env.GITHUB_EVENT_NAME;
+
+let msteams_webhook_url;
+let job_status;
+let event_id;
+let details;
+let account;
+let message;
+let url;
+
+
+async function parse_inputs() {
+    try {
+
+        // do not proceed if jobs was cancelled
+        job_status = core.getInput('job_status');
+        if (job_status.toUpperCase() === "CANCELLED") {
+            console.log("Job was cancelled, no notification will be sent")
+            process.exit(0);
+        }
+
+        msteams_webhook_url = core.getInput('webhook_url');
+        event_id = core.getInput('event_id');
+
+        const event_payload_data_text =  JSON.stringify(require(event_payload));
+        const event_payload_data = JSON.parse(event_payload_data_text);
```

Comment:
I'm not sure why on the previous line you `stringify` and then immediately afterwards `parse` here?
couldn't you reduce these two lines to:

```js
const event_payload_data = require(event_payload)
```

regardeless, I'd do this `require` outside of this function - especially if you want to keep it pure.

Reply:
These lines have been removed as we are now using yup schemata and validate/cast.